app/models/post_mover.rb

```diff
@@ -311,6 +419,28 @@ def delete_post_replies
     SQL
   end
 
+  def create_temp_post_timings
+    DB.exec("DROP TABLE IF EXISTS temp_post_timings") if Rails.env.test?
+    DB.exec <<~SQL
+      CREATE TEMPORARY TABLE temp_post_timings ON COMMIT DROP
+        AS (
+          SELECT pt.topic_id, mp.new_post_number as post_number, pt.user_id, pt.msecs
+          FROM post_timings pt
+          JOIN moved_posts mp
+            ON mp.old_topic_id = pt.topic_id
+              AND mp.old_post_number = pt.post_number
+              AND mp.old_topic_id = mp.new_topic_id
+        )
+    SQL
+  end
+
+  def insert_from_temp_post_timings
```

Comment:
@gschlager Thank you for your review!

I've renamed both methods and added a comment with some clarification, please let me know if it's enough, I can elaborate it further if it's still not clear or if you have any other suggestion.

For context, this temporary table was introduced with [this commit](https://github.com/discourse/discourse/pull/21374/commits/c4757ac1d435db69fda538e9b3c53de43b21fe85) to avoid updating all the shifted (same-topic posts that only moved positions) `post_timings` one by one: we copy the `post_timings` for the shifted posts with the `new_post_number` then re-insert them after the existing `delete_invalid_post_timings` method removes the `post_timings` for all `moved_posts`.


Thank you.